controllers/trainingjob/trainingjob_controller.go

```diff
@@ -185,8 +191,24 @@ func (r *Reconciler) reconcileTrainingJob(ctx reconcileRequestContext) error {
 		break
 
 	case sagemaker.TrainingJobStatusCompleted:
-		if err = r.addModelPathToStatus(ctx); err != nil {
-			return r.updateStatusAndReturnError(ctx, ReconcilingTrainingJobStatus, "", errors.Wrap(err, "Unable to add model path to status"))
+		if len(ctx.TrainingJob.Status.ModelPath) == 0 {
+			if err = r.addModelPathToStatus(ctx); err != nil {
+				return r.updateStatusAndReturnError(ctx, ReconcilingTrainingJobStatus, "", errors.Wrap(err, "Unable to add model path to status"))
+			}
+		}
+		// debug rule runs asynchronously with trainingjob. In some cases debug rule
+		// lags behind the trainingjob. Hence Training has to keep reconciling until
+		// its debug rule has completed.
+		// This breaks a bit k8s experience, since describe output will change evel
+		// if trainingjob has been completed.
+		// Its hard to populate the debug rules in `kubectl get` since not every trainifjccinlfg
+		// job will have debug rule and  column prints are not supported conditionally.
+		for _, debugRuleJob := range ctx.TrainingJob.Status.DebugRuleEvaluationStatuses {
```

Comment:
I think we'll want to update these statuses when the TrainingJob is InProgress as well. Why are we only updating debug rule statuses when TrainingJob is completed?

Reply:
Correct me if I am wrong @RedbackThomson The idea here is every requeue will update status of  debug job as well. So when training job is in progress the debug job status will be updated too, since requeue will execute that code path (I should link here which code path ?) as well, which will not happen if the training job has stopped.

Reply:
This code is only executed if the job status is `Completed`. For `InProgress` we would need to add it after the switch block. The reason I put `ModelPath` in here is because that field is only present when the status is `Completed`.

Reply:
I think that the code that updates the debug `RuleEvaluationStatus` should always execute if the TrainingJob exists, regardless of TrainingJob status. So, before the switch statement. I can't think of any good reason to not do so.